src/PackageHelper.js

```diff
@@ -0,0 +1,193 @@
+/**
+    Licensed to the Apache Software Foundation (ASF) under one
+    or more contributor license agreements.  See the NOTICE file
+    distributed with this work for additional information
+    regarding copyright ownership.  The ASF licenses this file
+    to you under the Apache License, Version 2.0 (the
+    "License"); you may not use this file except in compliance
+    with the License.  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+*/
+
+const fs = require('fs-extra');
+const detectIndent = require('detect-indent');
+const detectNewline = require('detect-newline');
+const writeFileAtomic = require('write-file-atomic');
+
+const DEFAULT_INDENT = 2;
+const CRLF = '\r\n';
+const LF = '\n';
+
+// Taken from https://github.com/npm/cli/blob/ab3c62aa83375a7d4a21e396129ae5679ee86646/lib/utils/stringify-package.js
+function stringifyPackage (data, indent, newline) {
```

Comment:
Why not extract stringify-package into its own package (with NPM license)?

Reply:
I don't know that I feel entirely comfortable taking ownership of this as its own module, given that I didn't write any of it, but I'll ping some of the npm folk and see if they'd be interested in doing that. I'm happy to do the work of pulling it into its own module, but I'd prefer they own it.

Reply:
Very thoughtful of you @dpogue. I think the ultimate solution is that they would extract stringify-package.js into its own package and reuse it within npm itself. Otherwise I guess you may as well do it yourself and offer shared ownership to the npm folk:)